### PR TITLE
Add switches (authentication collections) to support files

### PR DESCRIPTION
**Pull request by devhl-labs**
Added the authentication collections to the support files. I have a use case for it in my template where I need to know for a given apiInfos if it has a given authentication method.  Also standardized the collections provided. Before only OAuth methods were made available. It didn't make sense to not include other types as well.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java [hunk 1]
```diff
@@ -830,6 +812,36 @@ private void generateSupportingFiles(List<File> files, Map<String, Object> bundl
         return bundle;
     }
 
+    void addAuthenticationSwitches(Map<String, Object> bundle){
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java:815]: Suggestion: add a docstring explaining what this function does.

**Comment by devhl-labs** [comment]
Like my other two PR's these failed checks don't seem right.

**Comment by wing328** [comment]
cc @OpenAPITools/generator-core-team as it impacts the default generator/codegen.

**Comment by devhl-labs** [comment]
Merged master again hoping that the checks pass and that changes to samples go away. Samples should be updated via other pr's. I don't think anything here will change the samples.

**Comment by devhl-labs** [comment]
@wing328 @https://github.com/orgs/OpenAPITools/teams/generator-core-team Can I get some feedback here please?

**Comment by wing328** [comment]
Will try to review later today or this week

**Comment by wing328** [comment]
LGTM. Sorry for the delay in reviewing the enhancement.

**Pull request merged**